Shadowblade_main.lua

```diff
@@ -0,0 +1,330 @@
+object.nSoulUp = 20
+
+
+object.nGargantuanUseBonus = 16
+object.nFeintUseBonus = 12
+object.nSoulUseBonus = 10
+
+
+--ability use gives bonus to harass util for a while
+function object:oncombateventOverride(EventData)
+	self:oncombateventOld(EventData)
+	
+	local nAddBonus = 0
+	if EventData.Type == "Ability" then
+			
+		if EventData.InflictorName == "Ability_ShadowBlade1" then
+			nAddBonus = nAddBonus + object.nGargantuanUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade2" then
+			nAddBonus = nAddBonus + object.nFeintUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade3" then
+			nAddBonus = nAddBonus + object.nSoulUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade4" then
+			nAddBonus = nAddBonus + object.nSoulUseBonus
+		end
+	end
+	
+	if nAddBonus > 0 then
+		--decay before we add
+		core.DecayBonus(self)
+		core.nHarassBonus = core.nHarassBonus + nAddBonus
+	end
+end
+object.oncombateventOld = object.oncombatevent
+object.oncombatevent 	= object.oncombateventOverride
+------------------------------------------------------
+--            CustomHarassUtility Override          --
+-- Change Utility according to usable spells here   --
+------------------------------------------------------
+-- @param: IunitEntity hero
+-- @return: number
+local function CustomHarassUtilityFnOverride(hero)
+    local nUtil = 0
+	 
+    if skills.abilGargantuan:CanActivate() then
+        nUtil = nUtil + object.nGargantuanUp
+    end
+ 
+    if skills.abilFeint:CanActivate() then
+        nUtil = nUtil + object.nFeintUp
+    end
+ 
+    if skills.abilSoul:CanActivate() then
+        nUtil = nUtil + object.nSoulUp
+    end
+	
+
+
+    return nUtil
+end
+-- assign custom Harrass function to the behaviourLib object
+behaviorLib.CustomHarassUtility = CustomHarassUtilityFnOverride
+----------------------------------
+--	Shadowblade harass actions
+----------------------------------
+local function HarassHeroExecuteOverride(botBrain)
+	local bDebugEchos = false
+	
+	local unitTarget = behaviorLib.heroTarget
+	if unitTarget == nil or not unitTarget:IsValid() then
+		return false --can not execute, move on to the next behavior
+	end
+	
+	local unitSelf = core.unitSelf
+	local vecMyPosition = unitSelf:GetPosition()
+	
+	local vecTargetPosition = unitTarget:GetPosition()
+	local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition)
+	
+	local nLastHarassUtil = behaviorLib.lastHarassUtil
+	local bCanSee = core.CanSeeUnit(botBrain, unitTarget)	
+	
+	local bActionTaken = false
+	
+	local abilFeint = skills.abilFeint
+	local abilGargantuan = skills.abilGargantuan
+	local abilSoul = skills.abilSoul
+	local abilEssence = skills.abilEssence
+	
+--[[	-- bonus aggresion points for missing enemy HP  ---- Doesn't work
+    if bCanSee then
+		local nTargetHealth = unitTarget:GetHealthPercent()
+		nLastHarassUtil = nLastHarassUtil + (100 - nTargetHealth * 100)
+	end
+
+
+--]]
+
+	--Feint
```

Comment:
You may want to flesh Feint out a little, some of the higher-tier players make use of Feint's attack stealing capabilities (10/20/30/40%). Just adding a basic check looking for the enemy with the highest damage (or even just damage over a certain amount) could go a long way in winning a fight.